src/action/common/configure_nix.rs

```diff
@@ -98,17 +103,46 @@ impl Action for ConfigureNix {
         } = self;
 
         if let Some(configure_shell_profile) = configure_shell_profile {
+            let span = tracing::Span::current().clone();
+            let (span2, span3, span4) = (span.clone(), span.clone(), span.clone());
```

Comment:
This is kind of gross but we want to attach the span to each of the child tasks.

Reply:
Yup exactly why we can't. :(

We could certainly do that but I am not sure of value.